### PR TITLE
feat/enterpriseportal: validate and normalize instance domains

### PR DESCRIPTION
**Pull request by bobheadxi**
Validates domains on update (subscription and member) to ensure that we don't get anything too wonky.

## Test plan

Tests

### CODE REVIEW

File: lib/enterpriseportal/subscriptions/v1/v1_test.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+		domain:     "sourcegraph.com",
+		wantDomain: autogold.Expect(nil),
+	}, {
+		name:       "subdomain",
+		domain:     "foo.sourcegraph.com",
+		wantDomain: autogold.Expect(nil),
+	}, {
+		name:       "host with trailing slash",
+		domain:     "sourcegraph.com/",
+		wantDomain: autogold.Expect(nil),
```
Comment (unknwon) [inline lib/enterpriseportal/subscriptions/v1/v1_test.go:32]: shouldn't these 3 return valid?

Comment (bobheadxi) [reply lib/enterpriseportal/subscriptions/v1/v1_test.go:32]: Yes this needs a `-update` test run, but I wasn't able to get the test to compile due to some pgx issues (??)

Comment (unknwon) [reply lib/enterpriseportal/subscriptions/v1/v1_test.go]: Pushed a commit to fix that, 
```
go get github.com/jackc/pgconn
```

but it seem one of the test case failed 🤔

Comment (bobheadxi) [reply lib/enterpriseportal/subscriptions/v1/v1_test.go]: That's strange that it's missing from main 👁️

Comment (bobheadxi) [reply lib/enterpriseportal/subscriptions/v1/v1_test.go:32]: Anyway, fixed - should be working now!

**Comment by bobheadxi** [comment]
* **#63414** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/63414?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#63415** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/63415?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @bobheadxi and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by unknwon — APPROVED** [review]

**Comment by bobheadxi** [comment]
### Merge activity

* **Jun 21, 7:14 PM EDT**: @bobheadxi merged this pull request with [Graphite](https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/63415).

**Pull request merged**